modules/wsdata_field/src/Plugin/views/field/WSDataFieldsViews.php

```diff
@@ -48,13 +48,24 @@ public function render(ResultRow $values) {
     if ($entity->hasField($field_name)) {
       // Load the wsfield config entity.
       $wsfield_config = entity_load('wsfield_config', $field_name);
-      $wscall = entity_load('wscall', $wsfield_config->wscall);
 
       // Get the replacements.
       $replacements = is_array($wsfield_config->replacements) ? $wsfield_config->replacements : [];
-
+      $wsdata  = \Drupal::service('wsdata');
       // Create the call based on the wsfield configurations.
-      $return = $wscall->call(NULL, $replacements, $wsfield_config->data, array(), $wsfield_config->returnToken, array('node' => $entity));
+      $return = $wsdata->call(
+        $wsfield_config->wscall,
+        NULL,
```

Comment:
If NULL is likely to be the first argument, should be move it?

Reply:
Null should not be the first argument. This code was calling the WSCall entity directly instead of using the WSData service.

The method argument would be important to a RESTful connector, but we're not finding much use for it without one.  Likely we'll remove it in the next major version of WSData.

Reply:
Null should not be the first argument. This code was calling the WSCall entity directly instead of using the WSData service.

The method argument would be important to a RESTful connector, but we're not finding much use for it without one.  Likely we'll remove it in the next major version of WSData.